include/zephyr/arch/arm/aarch32/cortex_m/scripts/linker.ld

```diff
@@ -232,6 +232,16 @@ SECTIONS
 	. = ALIGN(_region_min_align);
 	_image_ram_start = .;
 
+#if defined(CONFIG_SEGGER_RTT_SECTION_FIXED_ADDR)
```

Comment:
That could be helpful also if a bootloader does not enable RTT at all (bootloader still may accidentally modify RAM area which application uses for the RTT). For the bootloader use-case, maybe we could allow to reserve memory without actually enabling the RTT support?

Reply:
I also wonder if we could somehow share required size of the RTT related section among bootloader and application (configuration must be aligned to ensure interoperability, otherwise we may see some strange errors). Maybe we could somehow propagate the information e.g. using SysBuild already

Reply:
I'm not sure Zephyr's build system + linker have anyway of putting it at the end of RAM? 

One way of achieving this is to ensure that the RTT section is the last thing placed in RAM as then we could use some of the values of the KConfig to figure out the size of the partition then set the (Location Counter)[https://ftp.gnu.org/old-gnu/Manuals/ld-2.9.1/html_chapter/ld_3.html#SEC10] to be  `RTT_START_ADDR = RAM SIZE - SIZE of RTT Block.`

The reason why this must be placed last is that if we move the LC forward we cannot move it back in GNUs LD. 

> Side note: ARMs scatter file/ARMs linker does have this ability.

Reply:
> That could be helpful also if a bootloader does not enable RTT at all (bootloader still may accidentally modify RAM area which application uses for the RTT). Alternatively, for the mentioned bootloader use-case, maybe we could allow to reserve memory without actually enabling the RTT support?

Agreed this would be useful ref. https://interrupt.memfault.com/blog/noinit-memory#retrofitting-a-noinit-region-when-you-cant-update-the-bootloader

Reply:
It would just be better to have a memory region setup in the dts and use that specifically for RTT. It should not just be assumed to be at the end of RAM, there are many reasons why it should not be placed there.

Reply:
@nordicjm : Sure a separate linker memory section for RTT would work fine thanks for the input 👍 Do you know how this can be achieved to section off the RAM by any chance?  Is there a way to define a node that would cut off the end of ram. 

IMO it should be user selectable where you want to place your RTT section but at least having it BEFORE and AFTER Application/Bootloader RAM would be two sensible options.

@giansta : https://docs.zephyrproject.org/apidoc/latest/devicetree__regions_8h.html Whatever Docs I found on this.

Reply:
I didn't want to add this at beginning, as the case for this PR (and for related issue #53544) was having a bootloader and app sharing the configuration.
But yes, the case with another bootloader where we want to keep the application logs between resets is also interesting. I've added a 2nd custom behaviour to use a DTS memory region that can be placed at end of RAM or at another address. I've tested it defining a region for my board. The two different choices are necessary, because of the snippet usage for the former one.
Anyway I wouldn't set it as default choice, because it needs a DTS definition, I kept the "normal" start of RAM option as default.